test/autoprefixer.test.js

```diff
@@ -641,7 +641,9 @@ describe('hacks', () => {
       'autoprefixer: <css input>:48:3: Autoplacement does not work ' +
         'without grid-template-rows property',
       'autoprefixer: <css input>:53:3: grid-auto-flow works only if grid-temp' +
-      'late-rows and grid-template-columns are present in the same rule'
+      'late-rows and grid-template-columns are present in the same rule',
+      'autoprefixer: <css input>:61:3: grid-gap works only if grid-temp' +
+      'late(-areas) is declared or when grid option is set to "autoplace"'
```

Comment:
> or when grid option is set to "autoplace"

That isn't quite right. It is misleading.

I think there needs to be different warnings depending on if the user has auto-placement enabled or not.

**If auto-placement is enabled:**

> grid-gap only works if grid-template(-areas) is declared or both both rows and columns have been declared and cells have not been manually placed inside the explicit grid

**If auto-placement is disabled:**

> grid-gap only works if grid-template(-areas) is declared

Reply:
@Dan503 I updated the warnings. Can you check it out?

Reply:
Yep the warnings are good now. 😊

I'm glad we added them. Autoplacement has made the situations where grid-gap is not supported a lot more complicated.